### PR TITLE
Support location in bq operator

### LINKED ISSUE #1088: Could not find BigQuery jobs in not US location
I'm trying to throw a query to BigQuery's dataset in asia-northeast1, but digdag can't find job's status.
It seem to be searching jobs with format ${PROJECT_NAME}:${JOB_ID} instead of ${PROJECT_NAME}:${LOCATION}:${JOB_ID}. isn't it?

### error logs
```
2019-04-16 21:41:43 +0900 [ERROR] (0095@[0:dataFlow]+main+test) io.digdag.core.agent.OperatorManager: Task +main+test failed.
digdag_1       | 404 Not Found
digdag_1       | {
digdag_1       |   "code" : 404,
digdag_1       |   "errors" : [ {
digdag_1       |     "domain" : "global",
digdag_1       |     "message" : "Not found: Job my-gcp-project:digdag_s0_p_my-digdag-project_w_main_t_950_a_13_6aa3ad0c-ebd2-4f78-a6b7-0c3e47957436",
digdag_1       |     "reason" : "notFound"
digdag_1       |   } ],
digdag_1       |   "message" : "Not found: Job my-gcp-project:digdag_s0_p_my-digdag-project_w_main_t_950_a_13_6aa3ad0c-ebd2-4f78-a6b7-0c3e47957436",
digdag_1       |   "status" : "NOT_FOUND"
digdag_1       | } (google json response)
```

### main.dig
```
test:
   bq>: queries/test.sql
   location: asia-northeast1
   dataset: my_dataset
```

### queries/test.sql
```
SELECT
  *
FROM
  `my_dataset.my_table`
```

### PR DESCRIPTION
**Pull request by tkawachi**
Set Job location for non-US location.

Fixes #1088

### CODE REVIEW

**Comment by tkawachi** [comment]
@yoyama can you review this PR?

File: digdag-standards/src/main/java/io/digdag/standards/operator/gcp/BaseBqJobOperator.java [hunk 1]
```diff
@@ -22,7 +22,8 @@
     protected TaskResult run(BqClient bq, String projectId)
     {
         BqJobRunner jobRunner = new BqJobRunner(request, bq, projectId);
-        Job completed = jobRunner.runJob(jobConfiguration(projectId));
+        String location = params.getOptional("location", String.class).orNull();
```
Comment (yoyama) [inline digdag-standards/src/main/java/io/digdag/standards/operator/gcp/BaseBqJobOperator.java:25]: I think using `Optional` is better

File: digdag-standards/src/main/java/io/digdag/standards/operator/gcp/BqClient.java [hunk 2]
```diff
@@ -170,11 +170,13 @@ void submitJob(String projectId, Job job)
...
     }
 
-    Job jobStatus(String projectId, String jobId)
+    Job jobStatus(String projectId, String jobId, String location)
             throws IOException
     {
         return client.jobs()
                 .get(projectId, jobId)
+                // newer version of google-api-services-bigquery-v2 has setLocation()
+                .set("location", location)
```
Comment (yoyama) [inline digdag-standards/src/main/java/io/digdag/standards/operator/gcp/BqClient.java:179]: I think calling only when `location` is present is better.

**Comment by yoyama** [comment]
@tkawachi Thank you for your contribution.
I left comments, please check them.
In addition, adding integration tests is preferred.
https://github.com/treasure-data/digdag/blob/master/digdag-tests/src/test/java/acceptance/td/BigQueryIT.java

File: digdag-standards/src/main/java/io/digdag/standards/operator/gcp/BaseBqJobOperator.java (hunk 1 above)
Comment (tkawachi) [reply digdag-standards/src/main/java/io/digdag/standards/operator/gcp/BaseBqJobOperator.java]: Updated.

File: digdag-standards/src/main/java/io/digdag/standards/operator/gcp/BqClient.java (hunk 2 above)
Comment (tkawachi) [reply digdag-standards/src/main/java/io/digdag/standards/operator/gcp/BqClient.java]: Done.

**Comment by tkawachi** [comment]
Thank you for your review, @yoyama.
I have added the integration test, but for the extract and load tests, the location of the BigQuery and GCS bucket must match. Currently, I suppose US bucket is set as GCS_TEST_BUCKET, but in addition to this, bucket of other location is needed.
If you can create the bucket for me, I will write an integrated test for extract and load. Since we used asia-northeast1 in other tests, a bucket for asia-northeast1 might be desirable.

**Comment by yoyama** [comment]
@tkawachi Thank you for your fix.
I created a bucket in asia region and set GCS_TEST_BUCKET_ASIA variable in CircleCI.
Could you confirm it ?

**Comment by tkawachi** [comment]
@yoyama I wrote tests for load and extract. It assumes that GCS_TEST_BUCKET_ASIA is in asia-northeast1, not asia (BigQuery can't be located in asia).  Where can I check the test result of CircleCI?

**Comment by yoyama** [comment]
@tkawachi I pushed your PR to digdag repo to run CircleCI.
It failed as follows.

https://app.circleci.com/pipelines/github/treasure-data/digdag/718/workflows/237633a7-4be4-4697-8e66-45195f1505b3/jobs/4699/parallel-runs/1?filterBy=ALL

Could you confirm it ?

File: digdag-tests/src/test/java/acceptance/td/BigQueryIT.java [hunk 3]
```diff
@@ -48,15 +48,7 @@
...
-import static acceptance.td.GcpUtil.BQ_TAG;
-import static acceptance.td.GcpUtil.GCP_CREDENTIAL;
-import static acceptance.td.GcpUtil.GCS_PREFIX;
-import static acceptance.td.GcpUtil.GCS_TEST_BUCKET;
-import static acceptance.td.GcpUtil.createDataset;
-import static acceptance.td.GcpUtil.createTable;
-import static acceptance.td.GcpUtil.datasetExists;
-import static acceptance.td.GcpUtil.listTables;
-import static acceptance.td.GcpUtil.tableExists;
+import static acceptance.td.GcpUtil.*;
```
Comment (yoyama) [inline digdag-tests/src/test/java/acceptance/td/BigQueryIT.java:51]: Could you avoid using `*` ?

**Comment by yoyama** [comment]
@tkawachi I add a minor comment. Could you confirm it ?

File: digdag-tests/src/test/java/acceptance/td/BigQueryIT.java (hunk 3 above)
Comment (tkawachi) [reply digdag-tests/src/test/java/acceptance/td/BigQueryIT.java]: updated

**Comment by tkawachi** [comment]
I updated the docs. I didn't change bq_ddl.md since it already mentions about location.

**Review by yoyama — APPROVED** [review]
LGTM.
Thank you for your contribution!

**Comment by toru-takahashi** [comment]
Hello @tkawachi 

Thank you for your great PR and contribution to digdag!
We are really appreciated with your efforts.

We would like to deliver some TD/Digdag swag to you as a token of our appreciation.
If you don't mind, could you fill your address in the following Google form? So that we can send to you.

https://forms.gle/37jhNBFJo9kyLvLx9

**Pull request merged**